pkg/clustermanager/reconcile_logicalcluster.go

```diff
@@ -53,9 +53,45 @@ func (c *controller) reconcileLogicalCluster(key string) error {
 	return nil
 }
 
-// TODO: the processAddLC needs to check whether the LogicalCluster has already
-// been added and processed, or whether this is actually an Update.
-func (c *controller) processAddLC(newCluster *lcv1alpha1.LogicalCluster) error {
+// processAddLC: process an added LC object
+func (c *controller) processAddLC(logicalCluster *lcv1alpha1.LogicalCluster) error {
+	logger := klog.FromContext(c.context)
+	var err error
+
+	switch logicalCluster.Status.Phase {
+	case lcv1alpha1.LogicalClusterPhaseEmpty:
+		// Either the client created a new logical cluster object and expects us to
+		// manage/create a physical cluster, or the discovery identified a new
+		// physical cluster.
+		return c.processNewLC(logicalCluster)
+	case lcv1alpha1.LogicalClusterPhaseInitializing:
+		// A managed cluster is being created by the provider. Need to wait for
+		// the cluster to be created at which point discovery will change the
+		// state to READY and update the cluster config.
+		return nil
+	case lcv1alpha1.LogicalClusterPhaseNotReady:
+		// Discovery noticed that an cluster has disappeared
+		// TODO: The current logic is missing a finalizer for managed logical cluster
+		return c.deleteLC(logicalCluster)
```

Comment:
Please add the logic of the conditions - isn't the above only for unmanaged ?

Reply:
I see it's in the function. Do we need function can't you just add the if and function call to here ?

Reply:
I thought it was cleaner to have the logic outside the switch, but you're probably right.  I moved it in.